src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -247,50 +326,95 @@ unique_ptr<CatalogEntry> TableCatalogEntry::AddColumn(ClientContext &context, Ad
 	}
 	Binder::BindLogicalType(context, info.new_column.type, schema->name);
 	info.new_column.oid = columns.size();
-	create_info->columns.push_back(info.new_column.Copy());
+	info.new_column.storage_oid = storage->column_definitions.size();
+	if (info.new_column.Generated()) {
+		info.new_column.storage_oid = EXCLUDED_FROM_STORAGE_ID;
+	}
+	// Add the dependencies of the generated column
+	if (info.new_column.Generated()) {
+		vector<string> referenced_columns;
```

Comment:
Move all binding and checking of dependents to the binder

Reply:
I'm a little blanking on what we discussed in regards to where the ColumnDependencyManager should live

As far as I gather, inside the CreateTableInfo we have all the data for the new table entry, with no context as to which ones are new.
But I think it's best to have the dependency mappings persist and not do unnecessary recreation, the only place I can do that is when creating the CreateTableInfo, because that happens in TransformCreateTable and in the TableCatalogEntry::[Add|Remove|Rename]Column methods

Reply:
I think because the dependencies aren't serialized, we have to recreate the dependencies inside of BindCreateTable anyways so I could move the dependency manager to BoundCreateTableInfo, or I should serialize the dependencies

Reply:
Ended up filling the dependency manager as part of BindCreateTableInfo, this also removed the need for the binding I was doing in the transformer phase.
To further clean up the need to do any premature binding, I have opted to make qualified names inside of generated expressions an error.